File: cases/taint/c/array-element-positive/array_element_positive.c
int dfb_source(void) { // DFB-SOURCE: array-element-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: array-element-sink

void run(void) {
    int values[2];
    values[0] = dfb_source(); // DFB-WITNESS: array-element-store
    values[1] = 0;
    dfb_sink(values[0]);
}

File: cases/taint/php/array-element-positive/array_element_positive.php
<?php
function dfb_source(): string { // DFB-SOURCE: array-element-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: array-element-sink

function run(): void {
    $values = ["tainted" => "clean", "clean" => "clean"];
    $values["tainted"] = dfb_source(); // DFB-WITNESS: array-element-store
    $values["clean"] = "clean";
    dfb_sink($values["tainted"]);
}
